### DISCUSSION TITLE
Guide for 90455e? 6 wire egg from 2020-21-7

### DISCUSSION THREAD

**Discussion by NapTight** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#534] [Q&A]
Hi. Sorry I'm not very experienced with GitHub and these kinds of projects. I tried searching but didn't find a match. Can someone point me towards a guide or help me with my search skills? Obviously I'm trying to add wifi and control to a Lay-z-spa. Model 90455e, 6 wire, egg, from 2020-21-7

**Comment by visualapproach** [comment]
What are you trying to find? On the main github page you will find the manual pdf. You won't find this model listed there because I've never seen it before. Maybe it's compatible with one of the 6-wire models supported.

**Reply by NapTight** [reply]
Thanks so much for your reply! See, I didn't even know there was a main page or manual. So I'll have a look there. I was hoping to find a guide with this exact model so I didn't have to "figure anything out." I'm pretty handy but slow to understand the nuances of these electronics projects. The name for this model is the Coleman Saluspa Miami. Do you happen to know if these are compatible or incompatible with the Bestway Lay-Z-Spa projects?

**Reply by visualapproach** [reply]
I don't KNOW until you report that it worked and that you had fun building it :) 
I know it has been working with some variants of Coleman saluspa but can't be sure of this.
Wish you luck if you try

**Reply by NapTight** [reply]
Well thanks again for replying! I'm reading through the excellent documentation now. I promise to have fun trying this project and to buy you a coffee 🫡

**Reply by NapTight** [reply]
Just to confirm this project is extremely cool and fun! It's also impressively robust and detailed. And to confirm it works with the model listed in this thread description. I used a D1 mini and [this guide](https://antanaitis.lt/2022/09/07/wi-fi-on-your-lay-z-spa-make-it-easy-to-remote-control/) to help me with wiring and configuration. If a dummy like me can do it anyone can.

**Reply by NapTight** [reply]
Beer money is on the way!

**Reply by visualapproach** [reply]
Awesome! Thank you for the nice review. And the beer 🍺 cheers mate

**Comment by NapTight** [comment]
Hey, sorry to be a bother. I can't get my spa to not shut itself off every few days. I attached a screenshot of my spa configuration. I figured setting the pump to on, the heater t
![Screenshot_20230621_131316_Edge](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/41939731/7889ed49-102b-4e20-b554-23f6d39d3fd4)
o on and the temp to 104F it should reset itself daily. But it will turns off.

**Reply by NapTight** [reply]
I should say I found the tub cold and the controller says "end"

**Reply by visualapproach** [reply]
Hey, what do you mean the controller says "end"?
I don't have time right now to dig into it, but I think two of your queued commands are redundant. 
Set target and set heat should be sufficient.

**Reply by NapTight** [reply]
This particular model shuts off every 72 hours I believe. The control screen on the pump flashes "end" and the temperature once it has shut off. This is the only thing I really want to use your Wi-Fi remote to fix. Where I live the weather is nice year round and we have solar electricity, so we just wanted to stay hot all the time. 

You are right. There is a redundant command in there. Maybe that's the problem?

**Reply by visualapproach** [reply]
I'm not jealous at all! "Nice weather year round"... well if you like a short summer and horizontal cold rain the rest of the year I could say the same.
May I ask where in the world you live?

Funny about the pump ending. It said in the manual for my pump that it should shut off after some days but it never did. When it says "end" can you then start it again by switching pump or heat on? In that case I guess you could catch the display characters, looking for "end" and then toggle the heat again. Or just trigger a heat command if not on. Do you have a home automation system? It could be done there. Or else maybe repeat the command queue command "heat on" every hour.

**Comment by NapTight** [comment]
Ah thank you. I will definitely try these. Like I said, I am pretty inexperienced with these projects. I just use the basic smart life app that works with the few items in my house. Or I use the proprietary apps for things like the ecobee thermostat. I would be interested in setting up automation through one hub. I know there's some content here discussing options for that.

I'm shocked at how much variance there is with these hot tubs. I didn't think this one was unique, but sounds like they're all a little quirky.

Sorry, I did not mean to make you jealous or brag. But I live in Ventura California. It is a beach town and really quite lovely. These are the average temps in Fahrenheit
![Screenshot_20230622_104508_Chrome](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/41939731/98938b27-ad25-425c-a3eb-fd75d273a171)

**Reply by visualapproach** [reply]
I know you didn't 😊
Are you all pet detectives? 😂
(Like Ace Ventura)

**Comment by NapTight** [comment]
Hahaha that's a good one! I did meet Jim Carey. He used to live close by. 

Allllrighhhty then!

**Reply by visualapproach** [reply]
Now you meant to make me jealous 😉! That's so cool! Spank you very much 😁

**Reply by NapTight** [reply]
Haha yes, I was. 😅

I have a similar result but not exactly the same. 

I set target at 104f and heat at 1 every 86400 sec. 

Now after 72 hours the spa is no longer heating, not does it say "end" as before. 

According to the GUI and the pump display the pump and heater are on, but the actual water temp is declining. So I think the spaconfig worked but the controller is still about to abort the heating after 72 hrs.

**Reply by visualapproach** [reply]
@NapTight okay, so how do you get it running again?

**Reply by NapTight** [reply]
Thank you again for helping me troubleshoot. I haven't tested all of the possible ways. Prior to performing your amazing project I would simply press the unlock button, the power button to shut down the spa pump, the power button again to turn on the spa, press the heat button, which turns on the heater and pump and finally verify the temp is set correctly.

I'm not sure if there's a way to cycle power to the whole unit using your configuration tool. But, I had the idea today to set the heat plus pump to zero then 1 minute later set them to one. I never tried resetting it using just the heat and/or pump buttons when it automatically shut down. I think maybe the reason why the daily configuration to turn the pump and heater on failed is because they hadn't been turned off for 72 hours. Anyways, what do you think?

**Reply by visualapproach** [reply]
That sounds plausible but darn it, I intentionally removed the power and lock buttons from the UI/command queue because it was just a hassle, but of course they had to make a pump that needs it. As you said, I don't see the point in all these different models. I'll see what I can do, but it will take a while due to me beeing too busy atm.

**Reply by NapTight** [reply]
Hi. Just want to confirm it worked!!! All I needed to do was set the pump and heat to cycle off before cycling them back on 1 minute later. Performing this daily has kept it running for a couple weeks now!! Thank you so much again!

**Reply by visualapproach** [reply]
Good work finding that out! Thx
